### PR TITLE
Make assets step in Dockerfile depend on locales, for jsi18n files

### PR DESCRIPTION
**Pull request by diox**
Fixes https://github.com/mozilla/addons/issues/15048

### Description

This fixes missing translations from our jsi18n files in our production images, which was happening because locales and assets were built in parallel.

### Context

We pregenerate [jsi18n](https://docs.djangoproject.com/en/5.1/topics/i18n/translation/#module-django.views.i18n) like regular static assets for performance reasons. That means the following needs to happen, for each locale:
- Our own `djangojs.mo` must be generated
- Then `manage.py generate_jsi18n_files` must be called
- Then `collectstatic` must be called

If one step is not done in this order, then the file will either be missing completely or missing our translations (only including django's, which are shipped out of the box).

There is a better way to fix this by completely moving all the bits `jsi18n` generation to its own step, and calling `collectstatic` at the end of the build, but that's a larger refactor and we need to fix this bug now.

### Testing

There was already an automated test about this, `test_generate_jsi18n_files`, but unfortunately it's passing because we aren't running tests against production images. And it's calling the command explicitly anyway, so it would hide the bug.

I've added a js test ran through `make test_static_assets`. The `Static Assets` job should not force a recompilation of assets or locales and still pass. Note that this test depends on `site-static/`, so if you're running with a local environment where `collecstatic` has not ran, it will fail.

To test manually, you need to inspect the `site-static/` and `static-build/` directories of a production image made from this branch. You can do it manually by running the image yourself, or use our abstractions meant for development, with some caveats:

- Run `make up DOCKER_TARGET=production COMPOSE_FILE=docker-compose.yml:docker-compose.ci.yml` with this branch. `web` and `worker` container will fail to go up because of https://github.com/mozilla/addons/issues/15058, ignore that by `Ctrl+C`ing the `make up` once it's reached the part where it's waiting for the containers to be healthy.
- Run `make shell` and verify that the `static-build/js/i18n/fr.js` and `site-static/js/i18n/fr.js` files contain one of our translations in French, such as `"Your add-on was validated with no errors or warnings"`. You can also check that a `site-static/js/i18n/fr.{hash}.js` file exist and is also up to date.
- Do not forget to reset your `.env` to drop the `DOCKER_TARGET` and `COMPOSE_FILE` before switching to something else. It's likely you don't want to stay in production mode.

### CODE REVIEW

**Comment by diox** [comment]
We have reverted https://github.com/mozilla/addons-server/pull/22710 because of unrelated breakage for local dev envs, and that revert breaks this fix. The locale is now correct in `static-build/` but wrong in `site-static/`...

**Comment by diox** [comment]
Something weird here locally:
- `make up COMPOSE_FILE="docker-compose.yml" DOCKER_TARGET="development"` ("normal" development setup) followed by `make run_js_tests`: tests **pass**
- `make up COMPOSE_FILE="docker-compose.yml:docker-compose.ci.yml"` followed by `make run_js_tests`: test I've added for `site-static/` **fails**
-  `make up COMPOSE_FILE="docker-compose.yml:docker-compose.ci.yml" DOCKER_TARGET=production` followed by `make run_js_tests`: test I've added for `site-static/` **fails**

Meanwhile in CI, [static assets tests are passing](https://github.com/mozilla/addons-server/actions/runs/11219539155/job/31185738064?pr=22717).

File: .dockerignore [hunk 1]
```diff
@@ -64,7 +64,6 @@ tmp/*
 docs
 private
 storage
-tests
```
[the remark this answers is not in the document]
Comment (diox) [reply .dockerignore:67]: We could mount them from the checkout clone somehow, but first I wanted to make them run and get them to be reliable.

**Comment by KevinMind** [comment]
> Something weird here locally:
> 
>     * `make up COMPOSE_FILE="docker-compose.yml" DOCKER_TARGET="development"` ("normal" development setup) followed by `make run_js_tests`: tests **pass**
> 
>     * `make up COMPOSE_FILE="docker-compose.yml:docker-compose.ci.yml"` followed by `make run_js_tests`: test I've added for `site-static/` **fails**
> 
>     * `make up COMPOSE_FILE="docker-compose.yml:docker-compose.ci.yml" DOCKER_TARGET=production` followed by `make run_js_tests`: test I've added for `site-static/` **fails**
> 
> 
> Meanwhile in CI, [static assets tests are passing](https://github.com/mozilla/addons-server/actions/runs/11219539155/job/31185738064?pr=22717).

This might be the problem.

```
COPY --chown=olympia:olympia locale/ ${HOME}/locale/
```

This is only copying the locale directory from the build context. If assets needs to depend on locales, then we should copy the directory from that stage by passing the `--stage=locales` flag.

**Comment by KevinMind** [comment]
Can you link the PR/issue where we decided to precompile the i18n js files? I would like to understand better how we got to the performance calculation/decision that we did.

It "feels" like generating on the fly would cause a slower request on the 1st request of a deployments lifecycle, but then the file is generated and it would be as fast as fully generating.. I'm not sure I understand why we need to "pre" generate.

I guess before we were just rendering on every request right?

**Comment by diox** [comment]
We were rendering dynamically, with heavy caching (and a cachebusting string depending on build ID). The main driver was simplifying SRE config to make it a regular static file, which has other benefits:making it a regular static file means no manual cachebusting, so cache stays fresh if the contents don't change from one deploy to the next, and pre-generating means no matter whether the state of the caches, the file is still served super quickly and efficiently. https://github.com/mozilla/addons-server/pull/10372

I don't want to go back to the dynamic way.

**Comment by diox** [comment]
We debugged this together and found that I had a stray volume causing my fix not to work locally. Once we got rid of that tests are passing. https://github.com/mozilla/addons-server/pull/22743 proves that without the fix, the test I've added here wouldn't pass.

**Review by KevinMind — APPROVED** [review]
Works.

**Pull request merged**